### PR TITLE
add data connector configuration to data_repository

### PR DESCRIPTION
**Pull request by khshah6**
add data connector configuration to data_repository

### CODE REVIEW

File: src/data_repository_manager.rs [hunk 1]
```diff
@@ -85,6 +85,7 @@ mod tests {
                 },
             }],
             metadata: meta.clone(),
+            data_connectors: vec![],
```
Comment (diptanu) [inline src/data_repository_manager.rs:88]: Add some data connectors here so that it is tested.

File: src/persistence.rs [hunk 2]
```diff
@@ -46,10 +46,34 @@ pub struct Extractor {
...
+    GoogleContact { metadata: Option<String> },
+    #[serde(rename = "gmail")]
+    Gmail { metadata: Option<String> },
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename = "content_type")]
+pub enum ContentType {
+    #[serde(rename = "document")]
+    Document,
```
Comment (diptanu) [inline src/persistence.rs:62]: May be add ContactInfo and Email as content type as well?

File: src/persistence.rs [hunk 3]
```diff
@@ -46,10 +46,34 @@ pub struct Extractor {
     pub extractor_type: ExtractorType,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename = "source_type")]
+pub enum SourceType {
+    #[serde(rename = "google_contact")]
+    GoogleContact { metadata: Option<String> },
```
Comment (khshah6) [inline src/persistence.rs:53]: todo: replace metadata with actual request parameters for GoogleContactApi

File: src/data_repository_manager.rs (hunk 1 above)
Comment (diptanu) [inline src/data_repository_manager.rs:88]: Add some data connectors here so that it is tested.

File: src/persistence.rs [hunk 4]
```diff
@@ -46,10 +46,34 @@ pub struct Extractor {
...
+    #[serde(rename = "gmail")]
+    Gmail { metadata: Option<String> },
```
Comment (khshah6) [inline src/persistence.rs:55]: todo: replace metadata with actual request parameters for gmail API

File: src/persistence.rs (hunk 2 above)
Comment (khshah6) [inline src/persistence.rs:62]: TODO: add different ContentTypes (ie: contact, email)

File: src/server.rs [hunk 5]
```diff
@@ -105,10 +107,60 @@ impl From<DataRepository> for ApiDataRepository {
...
+            ApiSourceType::Gmail { metadata } => SourceType::Gmail { metadata },
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename = "content_type")]
+pub enum ApiContentType {
+    #[serde(rename = "document")]
+    Document,
```
Comment (diptanu) [inline src/server.rs:131]: Email and Contact I assume?

**Review by diptanu — APPROVED** [review]
LGTM, small nits.

File: src/persistence.rs (hunk 2 above)
Comment (khshah6) [reply src/persistence.rs:62]: removing `ContentType` after offline conversation

File: src/server.rs (hunk 5 above)
Comment (khshah6) [reply src/server.rs:131]: removing ContentType after offline conversation

**Pull request merged**